src/odb/src/db/dbBTerm.cpp

```diff
@@ -921,4 +927,17 @@ void dbBTerm::staSetVertexId(uint32_t id)
   iterm->_sta_vertex_id = id;
 }
 
+void dbBTerm::setConstraintRegion(
+    const std::pair<Point, Point>& constraint_region)
+{
+  _dbBTerm* bterm = (_dbBTerm*) this;
+  bterm->_constraint_region = constraint_region;
+}
+
+std::optional<std::pair<Point, Point>> dbBTerm::getConstraintRegion()
+{
+  _dbBTerm* bterm = (_dbBTerm*) this;
+  return bterm->_constraint_region;
```

Comment:
If it is undefined you need to return an empty optional

Reply:
Now it should be complete.